### PR TITLE
Make parquet schema index type consistent

### PR DESCRIPTION
**Pull request by hyperbolic2346**
## Description
While working on parquet schema issue I noticed that the parent and child index didn't match. Discussion ensued and `size_type` was decided.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/parquet/parquet.hpp [hunk 1]
```diff
@@ -153,8 +155,8 @@ struct SchemaElement {
   // The following fields are filled in later during schema initialization
   int max_definition_level = 0;
   int max_repetition_level = 0;
-  int parent_idx           = 0;
-  std::vector<size_t> children_idx;
+  size_type parent_idx     = 0;
+  std::vector<size_type> children_idx;
```
Comment (vuule) [inline cpp/src/io/parquet/parquet.hpp:159]: I like how they were both wrong

**Review by vuule — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]

**Comment by ajschmidt8** [comment]
> /ok to test

@hyperbolic2346, you can prevent having to type `/ok to test` by configuring your commits to be signed as mentioned in the contributor link on the comment below

- https://github.com/rapidsai/cudf/pull/14256#issuecomment-1749349647

**Review by ttnghia — APPROVED** [review]

**Pull request merged**